Dockerfile.android

```diff
@@ -1,13 +1,30 @@
-FROM godot-fedora:latest
+ARG mono_version
+FROM godot-mono:${mono_version}
+ARG mono_version
 
 RUN dnf -y install scons java-1.8.0-openjdk-devel ncurses-compat-libs unzip which gcc gcc-c++ && \
+    mkdir sdk && cd sdk && \
     curl -LO https://dl.google.com/android/repository/sdk-tools-linux-4333796.zip && \
     unzip sdk-tools-linux-4333796.zip && \
     rm sdk-tools-linux-4333796.zip && \
     yes | tools/bin/sdkmanager --licenses && \
-    tools/bin/sdkmanager ndk-bundle 'platforms;android-23' 'build-tools;19.1.0' 'build-tools;28.0.3' 'platforms;android-28' 
+    tools/bin/sdkmanager ndk-bundle 'build-tools;28.0.3' 'platforms;android-28' 'cmake;3.10.2.4988404' && \
+    cd ..
 
-ENV ANDROID_HOME=/root/
-ENV ANDROID_NDK_ROOT=/root/ndk-bundle/
+ENV ANDROID_HOME=/root/sdk/
+ENV ANDROID_NDK_ROOT=/root/sdk/ndk-bundle/
+
+RUN dnf -y install git patch && \
+    git clone https://github.com/mono/mono --branch mono-${mono_version} --single-branch && \
```

Comment:
For the reference, while we use `--branch` here, we check out a git tag, so unless upstream deletes it and retags another commit, this should allow reproducible builds.
The `git submodule update --init` below should also update to the commit hashes referenced in the tagged mono commit, so they shouldn't change.